eth/tracers/firehose.go

```diff
@@ -169,6 +169,8 @@ type Firehose struct {
 	blockReorderOrdinalSnapshot uint64
 	blockReorderOrdinalOnce     sync.Once
 	blockIsGenesis              bool
+	captureBlock                bool
+	capturedBlock               *pbeth.Block
```

Comment:
We already have `testingBuffer` here that serves a similar purpose but for test, we should probably merge the two, propose me something for what we should do.